homeassistant/components/withings/config_flow.py

```diff
@@ -0,0 +1,242 @@
+"""Config flow for Withings."""
+from collections import OrderedDict
+import logging
+from typing import Optional
+
+import nokia
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import callback
+from homeassistant.util import slugify
+
+from . import const
+
+DATA_FLOW_IMPL = 'withings_flow_implementation'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def auth_callback_path(profile: str) -> str:
+    """Create an auth callback path."""
+    return '{}/{}'.format(const.AUTH_CALLBACK_PATH, slugify(profile))
+
+
+def auth_callback_name(profile: str) -> str:
+    """Create an auth callback name."""
+    return '{}:{}'.format(const.AUTH_CALLBACK_NAME, slugify(profile))
+
+
+@callback
+def register_flow_implementation(
+        hass,
+        client_id,
+        client_secret,
+        base_url,
+        profile
```

Comment:
Why do this instead of having people pick from a list of profiles during config flow ?